app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityTest.kt

```diff
@@ -939,21 +1080,67 @@ class NavigationDrawerActivityTest {
     fun inject(navigationDrawerActivityTest: NavigationDrawerActivityTest)
   }
 
+  // TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.
+  @Singleton
+  @Component(
+    modules = [
+      RobolectricModule::class,
+      TestDispatcherModule::class, ApplicationModule::class,
+      LoggerModule::class, ContinueModule::class, FractionInputModule::class,
+      ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+      NumberWithUnitsRuleModule::class, NumericInputRuleModule::class, TextInputRuleModule::class,
+      DragDropSortInputModule::class, ImageClickInputModule::class, InteractionsModule::class,
+      GcsResourceModule::class, GlideImageLoaderModule::class, ImageParsingModule::class,
+      HtmlParserEntityTypeModule::class, QuestionModule::class, TestLogReportingModule::class,
+      AccessibilityTestModule::class, LogStorageModule::class, CachingTestModule::class,
+      PrimeTopicAssetsControllerModule::class, ExpirationMetaDataRetrieverModule::class,
+      ViewBindingShimModule::class, RatioInputModule::class,
+      ApplicationStartupListenerModule::class, LogUploadWorkerModule::class,
+      WorkManagerConfigurationModule::class, HintsAndSolutionConfigModule::class,
+      FirebaseLogUploaderModule::class, FakeOppiaClockModule::class, PracticeTabModule::class,
+      DeveloperOptionsModule::class
+    ]
+  )
+  interface TestNoDebugApplicationComponent : ApplicationComponent {
+    @Component.Builder
+    interface Builder : ApplicationComponent.Builder
+
+    fun inject(navigationDrawerActivityTest: NavigationDrawerActivityTest)
+  }
+
   class TestApplication : Application(), ActivityComponentFactory, ApplicationInjectorProvider {
+    var disableDebugMode = false
+
     private val component: TestApplicationComponent by lazy {
       DaggerNavigationDrawerActivityTest_TestApplicationComponent.builder()
         .setApplication(this)
         .build() as TestApplicationComponent
     }
 
+    private val noDebugComponent: TestNoDebugApplicationComponent by lazy {
```

Comment:
What does this noDebugComponent mean here? Ideally the main component should be called `component` and the debug component should be called `debugComponent`.

Reply:
Actually, here most of the tests use `debugComponent` and only one test requires test application component without debug modules. So, I thought instead of creating `component` and `debugComponent` I should create a `component` with all the modules and a `noDebugComponent` without debug modules. The reason I named it `noDebugComponent` was because we cannot call it `prodComponent`. 

One thing we can do is, we can rename `TestApplicationComponent` to `TestDebugApplicationComponent` and `TestNoDebugApplicationComponent` to `TestApplicationComponent`. This will replace `component` with `debugComponent` and `noDebugComponent` to `component`. 

Or we can just rename  `noDebugComponent` to a proper name. I'm very bad with names so if you could suggest one then it will be really helpful :|

WDYT @Sarthak2601? Or if you have any other suggestions, please let me know. Thanks.

Reply:
I think we'll now need to shift from having 2 components to having two test files, so this won't make sense.

Reply:
I have separated the test files into `NavigationDrawerActivityProdTest.kt` and `NavigationDrawerActivityDevTest.kt`. PTAL @Sarthak2601 @BenHenning @anandwana001